command-expansion-server/src/app.ts

```diff
@@ -1,75 +1,293 @@
+    RETURNING (SELECT id FROM expansion_set_id);
+  `);
+
+  if (rows.length < 1) {
+    throw new Error(`PUT /expansion-set: No expansion set was inserted in the database`);
+  }
+  const id = rows[0].id;
+  console.log(`PUT /expansion-set: Updated expansion set in the database: id=${id}`);
+  res.contentType('json').status(200).send(JSON.stringify({ id }));
+  return;
+});
+
+app.get('/command-types/:dictionaryId(\\d+)', async (req, res) => {
+  const commandTypes = await getCommandTypes(db, parseInt(req.params.dictionaryId, 10));
+  res.contentType('text').status(200).send(commandTypes);
+  return;
+});
+
+app.get('/activity-types/:missionModelId(\\d+)/:activityTypeName', async (req, res) => {
+  const activityTypes = await getActivityTypes(graphqlClient, parseInt(req.params.missionModelId, 10), req.params.activityTypeName);
+  res.contentType('text').status(200).send(activityTypes);
+  return;
+});
+
+app.get('/commands/:expansionRunId(\\d+)/:activityInstanceId(\\d+)', async (req, res) => {
+  // Pull existing expanded commands for an activity instance of an expansion run
+  res.status(501).send('GET /commands: Not implemented');
+  return;
+});
+
+app.post('/expand-all-activity-instances/:simulationId(\\d+)/:expansionSetId(\\d+)', async (req, res) => {
+  const simulationId = parseInt(req.params.simulationId, 10);
+  const expansionSetId = parseInt(req.params.expansionSetId, 10);
+
+  const [
+    { missionModelId: simulationMissionModelId, activityInstances},
+    { commandDictionaryId, missionModelId: expansionSetMissionModelId, expansionMap },
+  ] = await Promise.all([
+    getActivityInstancesAndModelIdFromSimulationId(graphqlClient, simulationId),
+    getExpansionSet(db, expansionSetId),
+  ]);
+
+  if (simulationMissionModelId !== expansionSetMissionModelId) {
+    throw new Error(`POST /expand-all-activity-instances: Attempting to expand and activity instance from a simulation with a different mission model (${simulationMissionModelId}) than the specified expansion set (${expansionSetMissionModelId})`);
+  }
+  const missionModelId = simulationMissionModelId;
+
+  const commandTypes = await getCommandTypes(db, commandDictionaryId);
+
+  const activityInstanceCommandPromiseSettledResult = await Promise.allSettled(activityInstances.map(async (activityInstance) => {
+    const expansionLogic = expansionMap[activityInstance.type];
+    if (expansionLogic !== undefined) {
+      try {
+        const result: { commands: ReturnType<Command['toSeqJson']>[]} = await piscina.run({
+          expansionLogic: expansionLogic,
+          activityTypeName: activityInstance.type,
+          commandTypes,
+          activityTypes: await getActivityTypes(graphqlClient, missionModelId, activityInstance.type),
+          activityInstance,
+        })
+        return {
+          activityInstanceId: activityInstance.id,
+          commands: result.commands
+        };
+      } catch(e: any) {
```

Comment:
More of a question for my own TS learning, but is catching `any` exception an anti-pattern? Similar to how `catch (final Exception e)` is in Java. My reasoning comes from [this](https://stackoverflow.com/questions/2416316/why-is-the-catchexception-almost-always-a-bad-idea):
>when you catch all exceptions, you may get an exception that cannot deal with and prevent code that is upper in the stack to handle it properly.

Reply:
We can definitely specialize errors in the future for better error handling, but in the first cut, I think we're fine saying "Something went wrong while doing this"

Reply:
That's fair enough. I try to be as specific with exception catching as possible, even when prototyping since it's easy to forget to circle back to these things. This is re-throwing with the intention of *adding* additional context but it's really just re-throwing and *obfuscating* the original context since any exception is now an `InheritedError`.

Either way, not a big deal, but if this `e: any` can be more specific and takes just a minute it may be worthwhile to just get right now rather than later :)

Reply:
Yeah, this one especially is tricky because the worker will throw and the error is serialized + deserialized as just a base Error